### PR TITLE
feat: ability to update personal profile picture

### PR DESCRIPTION
**Pull request by Sboonny**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `main` branch of Chapter.

this aims to create, profile picture field which will be used by `Avatar` chakra component to display personal characteristics into profile picture. 

Will I attempt to block any type of images? nope, just adding the ability to upload an image, more than can be dealt with post MVP

the outcome 

image

### CODE REVIEW

File: client/src/components/PageLayout/Header.tsx [hunk 1]
```diff
@@ -78,6 +78,7 @@ export const Header: React.FC = () => {
 
   const goHome = () => router.push('/');
 
+  console.log(user?.image_url);
```
Comment (gikf) [inline client/src/components/PageLayout/Header.tsx:81]:
```suggestion
```

**Review by gikf — COMMENTED** [review]
I don't know if that's intended or not. When using transparent image (ie. https://cdn-icons-png.flaticon.com/512/149/149157.png), the background color that's set by the user name will still be used.

image image

**Comment by Sboonny** [comment]
It isn't intended, it's chakra doing good work.

That being said, do we want transparent image? I think it's harmful for UX, but we don't have users yet so we can have it 🤔 and test it

**Comment by Sboonny** [comment]
Stop it as 100%? I don't think so. 

But we can create an `::after` content (maybe a nice trending glass effect) that stay behind the image, if the image is transparent, then that `::after` content will show up instead (well technically the image will disappear and what behind it (::after) will show up).

Should I give that a go and try to override chakra background color?

**Comment by ojeytonwilliams** [comment]
I could be underthinking it, but how about setting the background to white if there's an image?

**Comment by Sboonny** [comment]
There are pages that use `avatar` component in them, mainly ChapterPage and EventPage, although this isn't affecting them yet. it should be DRYed and used on these pages. if we're going to DRY it, using background white won't work in UX aspect but the idea is on the mark 👍

**Comment by Sboonny** [comment]
I have reread this, and I was confused. I though we didn't want a transparent Image, but we do want transparent photo, I am adding them 👍

File: server/prisma/schema.prisma [hunk 2]
```diff
@@ -264,6 +264,7 @@ model users {
   updated_at       DateTime        @updatedAt
   id               Int             @id @default(autoincrement())
   name             String
+  image_url        String
```
Comment (ojeytonwilliams) [inline server/prisma/schema.prisma:267]:
```suggestion
  image_url        String?
```
Having this be nullable better reflects the fact that the image_url is optional. It also means that `image_url: ''` can be deleted everywhere.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

File: server/prisma/generator/factories/user.factory.ts [hunk 3]
```diff
@@ -19,27 +19,31 @@ const createUsers = async (
     email: 'foo@bar.com',
     name: 'The Owner',
     instance_role: { connect: { id: instanceRoles.owner.id } },
+    image_url: '',
```
Comment (ojeytonwilliams) [inline server/prisma/generator/factories/user.factory.ts:22]:
```suggestion
```

File: server/prisma/generator/factories/user.factory.ts [hunk 4]
```diff
@@ -19,27 +19,31 @@ const createUsers = async (
...
   };
   const owner = await prisma.users.create({ data: ownerData });
 
   const chapter1AdminData: Prisma.usersCreateInput = {
     email: 'admin@of.chapter.one',
     name: 'Chapter One Admin',
     instance_role: { connect: { id: instanceRoles.chapter_administrator.id } },
+    image_url: '',
```
Comment (ojeytonwilliams) [inline server/prisma/generator/factories/user.factory.ts:30]:
```suggestion
```

File: server/prisma/generator/factories/user.factory.ts [hunk 5]
```diff
@@ -19,27 +19,31 @@ const createUsers = async (
...
   };
   const chapter1Admin = await prisma.users.create({ data: chapter1AdminData });
 
   const chapter2AdminData: Prisma.usersCreateInput = {
     email: 'admin@of.chapter.two',
     name: 'Chapter Two Admin',
     instance_role: { connect: { id: instanceRoles.chapter_administrator.id } },
+    image_url: '',
```
Comment (ojeytonwilliams) [inline server/prisma/generator/factories/user.factory.ts:38]:
```suggestion
```

File: server/prisma/generator/factories/user.factory.ts [hunk 6]
```diff
@@ -19,27 +19,31 @@ const createUsers = async (
...
   };
   const chapter2Admin = await prisma.users.create({ data: chapter2AdminData });
 
   const bannedAdminData: Prisma.usersCreateInput = {
     email: 'banned@chapter.admin',
     name: 'Banned Chapter Admin',
     instance_role: { connect: { id: instanceRoles.chapter_administrator.id } },
+    image_url: '',
```
Comment (ojeytonwilliams) [inline server/prisma/generator/factories/user.factory.ts:46]:
```suggestion
```

File: server/prisma/generator/factories/user.factory.ts [hunk 7]
```diff
@@ -48,6 +52,7 @@ const createUsers = async (
     email: 'test@user.org',
     name: 'Test User',
     instance_role: { connect: { id: instanceRoles.member.id } },
+    image_url: '',
```
Comment (ojeytonwilliams) [inline server/prisma/generator/factories/user.factory.ts:55]:
```suggestion
```

File: server/prisma/generator/factories/user.factory.ts [hunk 8]
```diff
@@ -58,6 +63,7 @@ const createUsers = async (
       email: internet.email(),
       name: `${name.firstName()} ${name.lastName()}`,
       instance_role: { connect: { id: instanceRoles.member.id } },
+      image_url: '',
```
Comment (ojeytonwilliams) [inline server/prisma/generator/factories/user.factory.ts:66]:
```suggestion
```

File: server/src/app.ts [hunk 9]
```diff
@@ -120,6 +120,7 @@ export const main = async (app: Express) => {
             name: 'member',
           },
         },
+        image_url: '',
```
Comment (ojeytonwilliams) [inline server/src/app.ts:123]:
```suggestion
```

File: server/tests/fixtures/users.ts [hunk 10]
```diff
@@ -38,6 +38,7 @@ const baseUser: User = {
   created_at: new Date(),
   updated_at: new Date(),
   email: 'an@add.ress',
+  image_url: '',
```
Comment (ojeytonwilliams) [inline server/tests/fixtures/users.ts:41]:
```suggestion
```

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Pretty sure we don't need any of these.

File: server/prisma/generator/factories/user.factory.ts (hunk 3 above)
Comment (Sboonny) [reply server/prisma/generator/factories/user.factory.ts]: Oh 😅, they were an attempt to please typescript, is there a better way around that type error?🤔

**Comment by ojeytonwilliams** [comment]
@Sboonny I seem to have upset Cypress. Sorry about that!  I'll fix it shortly.

**Comment by Sboonny** [comment]
no worries, feel free to 👍

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1: Assuming the CI passes, :crossed_fingers:, this seems ready to me.

**Pull request merged**